test/groovy/DockerExecuteTest.groovy

```diff
@@ -1,61 +1,97 @@
 
 import org.junit.Before
+import org.junit.Rule
 import org.junit.Test
 
+import com.lesfurets.jenkins.unit.BasePipelineTest
+
+import util.JenkinsConfigRule
+import util.JenkinsLoggingRule
+import util.JenkinsSetupRule
+
 import static org.junit.Assert.assertEquals
 import static org.junit.Assert.assertTrue
 import static org.junit.Assert.assertFalse
 
-class DockerExecuteTest extends PiperTestBase {
+class DockerExecuteTest extends BasePipelineTest {
     private DockerMock docker
 
-    String echos
+    @Rule
+    public JenkinsSetupRule jsr = new JenkinsSetupRule(this)
+
+    @Rule
+    public JenkinsLoggingRule jlr = new JenkinsLoggingRule(this)
+
+    @Rule
+    public JenkinsConfigRule jcr = new JenkinsConfigRule(this)
 
     int whichDockerReturnValue = 0
 
+    def cpe
+    def examinee;
+
     @Before
-    void setUp() {
-        super.setUp()
+    void init() {
 
         docker = new DockerMock()
         binding.setVariable('docker', docker)
         binding.setVariable('Jenkins', [instance: [pluginManager: [plugins: [new PluginMock()]]]])
 
-        echos = ''
-        helper.registerAllowedMethod("echo", [String.class], { String s -> echos += " $s" })
         helper.registerAllowedMethod('sh', [Map.class], {return whichDockerReturnValue})
+
+        prepareObjectInterceptors(this)
+
+        cpe = loadScript('commonPipelineEnvironment.groovy').commonPipelineEnvironment
+        examinee = loadScript('dockerExecute.groovy').dockerExecute
     }
 
     @Test
     void testExecuteInsideDocker() throws Exception {
-        def script = loadScript("test/resources/pipelines/dockerExecuteTest/executeInsideDocker.groovy")
-        script.execute()
+
+        examinee.call(script: [commonPipelineEnvironment: cpe],
+                      dockerImage: 'maven:3.5-jdk-8-alpine') {
+            echo 'Body executed'
+        }
+
         assertEquals('maven:3.5-jdk-8-alpine', docker.getImageName())
         assertTrue(docker.isImagePulled())
         assertEquals(' --env http_proxy --env https_proxy --env no_proxy --env HTTP_PROXY --env HTTPS_PROXY --env NO_PROXY', docker.getParameters())
-        assertTrue(echos.contains('Inside Docker'))
+        assert jlr.log.contains('Body executed')
```

Comment:
why not using `assertTrue(jlr.log.contains('...'))` to make it consistent?

Reply:
done